lib/fluent/plugin/in_syslog.rb

```diff
@@ -181,12 +181,14 @@ def start_tcp_server(tls: false)
           if octet_count_frame
             while idx = buffer.index(delimiter, pos)
               num = Integer(buffer[pos..idx])
-              pos = idx + num
-              msg = buffer[idx + 1...pos]
-              if msg.size < num - 1
-                pos = pos - num - num.to_s.size
+              idx = idx + delimiter_size
+              msg = buffer[idx, num]
+              if msg.size != num
                 break
               end
+              pos = idx + msg.size
+              # syslog might append a \n as an extra frame delimiter
+              msg.chomp!
```

Comment:
How can we handle if the user adds `\n` at the end of msg intentionally? 
In the situation, it shouldn't trim `\n` from msg.

Reply:
I was limiting this change to preventing the removal of characters other than `\n`. I could add that as a configuration option to control if the chomp should be called.

Reply:
I looked into this more and currently in the default parser configuration chomping at the in_syslog level has no impact because parse_syslog strips off trailing `\n`.

However, that is only part of the story. Until https://github.com/fluent/fluentd/pull/2767 lands if RFC5424 is selected as the format for parser_syslog messages it will fail to parse unless in_syslog does a chomp on the incoming messages.

There also is a second issue, without this change octet counted transport always strips off the last character of a message (incorrectly presumed to be a delimiter) while the delimited transport does not strip off the delimiter! So if chomp is off by default then it will be a behavior change for octect counted transport. If chomp is on by default then it will be a behavior change for delimited transport. In light of that I would personally would keep chomp off by default as it because it will impact fewer users and this change is supposed to address that fact that octect counted transport is currently incorrect.